src/client/dfuse/il/int_posix.c

```diff
@@ -798,6 +799,7 @@ child_hdlr(void)
 {
 	int rc;
 
+	daos_eq_lib_reset_after_fork();
```

Comment:
in this case you are only reinitializing cart and the internal event.
so what about everything else that daos_init does?

Reply:
also im questioning what is this solving.. is fork now supported with this?
there are more fundamental issues with fork i believe, no?

Reply:
Right. There could still be issues somewhere else due to global variables. I have not run tests thoroughly about fork(). I ran fio and another small piece of code testing DFS both in parent and child processes after fork(). They can run without noticing issue now. They previously got time out.

For fio with ioil, the child processes only access DFS in read/write through the context newly created. There was no timeout issue there. I did modify fio code to let child processes access DFS with the default context inherited from the parent process, then I observed the same timeout issue when parent process accesses DFS with the default context later. With daos_eq_lib_reset_after_fork() called, the timeout issue is fixed.

I agree we need to check other code in daos_init() too before claiming the support of fork().

Reply:
OK, i am fine if you add this change, although it seems unnecessary as fork support isn't happening.
FIO support is already there without this change anyway..

Reply:
if you noticed, my comment is in the ioil section not the pil4dfs one ;-)

Reply:
anyway, not requesting a change